frontend/src/Frontend/ChainwebApi.hs

```diff
@@ -501,3 +546,47 @@ getChainwebStats nc evt = do
                 (constDyn $ mkDataUrl dh)
         txResp <- go evt
         return $ r2e <$> txResp
+
+getTransfers
+    :: forall t m. (TriggerEvent t m, PerformEvent t m,
+        HasJSContext (Performable m), MonadJSM (Performable m))
+    => NetConfig
+    -> Dynamic t (QParam Limit)
+    -> Dynamic t (QParam Offset)
+    -> Dynamic t (Either Text Text) -- Account Name
+    -> Dynamic t (QParam Text) -- Token
+    -> Dynamic t (QParam ChainId)
+    -> Dynamic t (QParam NextToken)
+    -> Event t ()
+    -> m (Event t (Either TransferError ([AccountDetail], Maybe NextToken)))
+getTransfers nc lim off account token chain nextToken evt = do
+    case _netConfig_dataHost nc of
+      Nothing -> return never
+      Just dh -> do
+        let ((_ :<|> _ :<|> _  :<|> _ :<|> _ :<|> go) :<|> _) =
+              clientWithOpts chainwebDataApi
+                (Proxy :: Proxy m)
+                (Proxy :: Proxy ())
+                (constDyn $ mkDataUrl dh)
+                transferOpts
+	trResp <- go account token chain lim off nextToken evt
+        return $ go_ <$> trResp
+  where
+    go_ = \case
+        ResponseSuccess _ a _ ->
+                Right (getResponse a, getNextHeader a)
+        ResponseFailure _ txt xhrResponse
+                | _xhrResponse_status xhrResponse /= 200 -> Left $ NonHTTP200 (_xhrResponse_status xhrResponse)
```

Comment:
Seems to me like we're swallowing the most useful kind of error message here. For example, if we get a `400 invalid next token` response, this will only become `NonHTTP200 400` and we'll lose the message. Why not just have `Left $ BadResponse txt` for all status codes?